PostAppTest/CommentTest.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Moq;
+using Server.Controllers;
+using Server.Models;
+using Server.Repository;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace PostAppTest
+{
+    public class CommentTest
+    {
+        [Fact]
+        public void GetCommentsByPostId_Returns_All_Comments()
+        {
+
+            var id = 1;
+            // Arrange
+            var mockRepo = new Mock<ICommentRepository>();
+
+            mockRepo.Setup(repo => repo.GetCommentsByPostId( id))
+            .Returns(GetTestComments());
+
+            var controller = new CommentsController(mockRepo.Object);
+
+            var result = controller.GetCommentsByPostId(1);
+
+            Assert.NotNull(result);
```

Comment:
As this will return the list. Check comments count also.  Update same for other test cases.